be/dtm0_log.h

```diff
@@ -0,0 +1,522 @@
+ * @param log Pointer to a log structure that has been previously allocated.
+ *
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_free(struct m0_be_dtm0_log **log);
+
+/**
+ * For performing an operation on a persistent log, we need to take the
+ * appropriate number of be transaction credit.
+ *
+ * @pre None
+ * @post None
+ *
+ * @param op Operation code for the type of dtm0 log operation that we want
+ *        to perform. @see @ref m0_be_dtm0_log_credit_op.
+ * @param txd A valid dtm0 tx descriptor that we want to store in the log
+ *        record.
+ * @param payload The payload is an opaque structure for dtm0 log to store
+ *        in the log record.
+ * @param seg The be segment on which the log resides.
+ * @param A pointer to a valid m0_dtm0_log_rec.
+ * @param accum This contains the number of credits required for performing
+ *        the operation.
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_credit(enum m0_be_dtm0_log_credit_op op,
+                                       struct m0_dtm0_tx_desc	    *txd,
+                                       struct m0_buf		    *payload,
+                                       struct m0_be_seg             *seg,
+                                       struct m0_dtm0_log_rec	    *rec,
+                                       struct m0_be_tx_credit       *accum);
+
+/**
+ * This routine is used to create a dtm0 persistent log.
+ *
+ * @pre tx != NULL
+ * @pre seg != NULL
+ * @post None
+ *
+ * @param tx handle to an already opened transaction.
+ * @param seg be segment on which to create the log.
+ * @param out variable in which to store the pointer to the newly created
+ *        log.
+ * @return 0 on success, anything else is a failure.
+ */
+M0_INTERNAL int m0_be_dtm0_log_create(struct m0_be_tx        *tx,
+                                      struct m0_be_seg       *seg,
+                                      struct m0_be_dtm0_log **out);
+
+/**
+ * This routine is used to destroy a previously created persistent dtm0 log
+ *
+ * @pre tx != NULL
+ * @pre *log != NULL
+ * @post None
+ *
+ * @param tx handle to an already opened transaction.
+ * @param log Pointer to the dtm0 log which we wish to destroy
+ * @return None
+ */
+M0_INTERNAL void m0_be_dtm0_log_destroy(struct m0_be_tx        *tx,
+                                        struct m0_be_dtm0_log **log);
+
+/**
+ * This routine is used to update a record in the dtm0 log. If the record
+ * does not exist it will be inserted.
+ *
+ * @pre payload != NULL
+ * @pre m0_be_dtm0_log__invariant(log)
+ * @pre m0_dtm0_tx_desc__invariant(txd)
+ * @pre m0_mutex_is_locked(&log->dl_lock)
+ * @post None
+ *
+ * @param log Pointer to the dtm0 log which we wish to destroy
+ * @param tx handle to an already opened transaction.
+ * @param txd A valid dtm0 tx descriptor that we want to store in the log
+ *        record.
+ * @param payload The payload is an opaque structure for dtm0 log to store
+ *        in the log record.
+ * @return None
```

Comment:
No return? We need to update this comment or the code.

Reply:
Done. Updated the comment.